### ISSUE TITLE
Encapsulate Peer in an implementation-independent manner

### ISSUE THREAD

**Issue by ptesavol** [streamr-dev/network-old#80]
At the moment the Peer handling is very much dependent on Libp2p specific data structures even on the Logic layer. It is still undecided in which way the Peer will be handled in an implementation-independent manner. I see at leat two possible solutions here: 1. an implementation-independent Peer class that encapsulates all necessary information about a Peer, or 2. passing only peerId (string) cross-layer, and trying to minimise the information the higher layers need to know about peers.  

Which option should we go for in your opinion? Any other solutions come to mind?

**Comment by harbu** [comment]
Can you give us some concrete example(s) of this dependency?
 
The only one I can think of is the usage of `PeerInfo` object in logic layer. Two operations are needed are `getAddress` and `getId` (latter mostly for logging, currently).

Maybe we should create a wrapper to decouple logic-layer from libp2p _or_ just deal with string addresses in logic layer? (Of course the string addresses follow IPFS format, but the logic layer is not concerned with the contents of the string, just that it refers to a peer in the network)

**Comment by harbu** [comment]
Well I wrote almost the same thing as you, now on further inspection. 😅

**Comment by harbu** [comment]
Were there any other leaking abstractions?

**Comment by mirpo** [comment]
Let's postpone it until we'll have DHT and all needed functions for MVP. Now we're using only one address (for simplifying development process and debugging), but in Libp2p we can have multiple, so anyway we'll move away from using string addresses to some PeerID class

**Comment by ptesavol** [comment]
Well, this one leaks all the way from logic layer to the internals of LibP2P https://github.com/streamr-dev/network/blob/e26b6e82afec48e5e86b48a99b86dec15e03d31e/src/logic/Publisher.js#L9

But maybe this one should be made into a separate issue such as "make underlying layers private"

**Comment by ptesavol** [comment]
I think we could use string or integer peerId:s that do not necessarily even need to be long, cryptographic identifiers, but rather could just be indexes to the peers table on the connection layer. The main thing is that the upper layers should not need to care nor know what the peerIds are internally.

**Comment by harbu** [comment]
Not using libp2p, so closing.
